lib/streams/recordStream.js

```diff
@@ -8,6 +9,29 @@ var CleanupStream = require('./cleanupStream');
 var ValidRecordFilterStream = require('./validRecordFilterStream');
 var DocumentStream = require('./documentStream');
 
+/*
+ * Construct a suitable id prefix for a CSV file given
+ * its full filename and the base directory of all OA CSV files.
+ */
+function getIdPrefix(filename, dirPath) {
+  var heirarchy = [];
+  var current_level = filename;
+
+  // traverse the directory structure upward to the OA base directory,
+  // keeping track of directory names along the way
+  while (current_level !== '/') {
```

Comment:
this whole bit seems hard to follow. Maybe using something like this would be clearer?

`'/foo/bar/baz.csv'.split(require('path').sep);` results in `[ '', 'foo', 'bar', 'baz.csv' ]`
and then just iterate through the array